sdk/src/wallet/account/operations/balance.rs

```diff
@@ -275,17 +96,258 @@ impl Account {
+                                        // Amount for basic outputs isn't added to total_rent_amount if there aren't
+                                        // native tokens, since we can
+                                        // spend it without burning.
+                                        if output
+                                            .native_tokens()
+                                            .map(|native_tokens| !native_tokens.is_empty())
+                                            .unwrap_or(false)
+                                            && !context.account_details.locked_outputs.contains(output_id)
+                                        {
+                                            context.total_rent_amount += rent;
+                                        }
+                                    } else if output.is_nft() {
+                                        context.balance.required_storage_deposit.nft += rent;
+                                        if !context.account_details.locked_outputs.contains(output_id) {
+                                            context.total_rent_amount += rent;
+                                        }
+                                    }
+
+                                    // Add native tokens
+                                    if let Some(native_tokens) = output.native_tokens() {
+                                        context.total_native_tokens.add_native_tokens(native_tokens.clone())?;
+                                    }
+                                } else {
+                                    // only add outputs that can't be locked now and at any point in the future
+                                    context.balance.potentially_locked_outputs.insert(*output_id, true);
+                                }
+                            } else {
+                                // Don't add expired outputs that can't ever be unlocked by us
+                                if let Some(expiration) = output
+                                    .unlock_conditions()
+                                    .expect("output needs to have unlock conditions")
+                                    .expiration()
+                                {
+                                    // Not expired, could get unlockable when it's expired, so we insert it
+                                    if local_time < expiration.timestamp() {
+                                        context.balance.potentially_locked_outputs.insert(*output_id, false);
+                                    }
+                                } else {
+                                    context.balance.potentially_locked_outputs.insert(*output_id, false);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    // TODO address param or another method?
```

Comment:
No strong opinion here, single method is less code, two methods is probably clearer
But do we think a single address is sufficient or should we accept a Vec?

Reply:
I think I'll go with one method, easier for message interface etc

Reply:
I can do a Vec, sure